src/core/truthTable/truth_table.cpp

```diff
@@ -29,38 +29,45 @@ namespace syrec {
         return result;
     }
 
-    auto TruthTable::equal(TruthTable& tt1, TruthTable& tt2, bool equalityUpToDontCare) -> bool {
+    auto TruthTable::equal(TruthTable const& tt1, TruthTable const& tt2, bool equalityUpToDontCare) -> bool {
         if (!equalityUpToDontCare) {
             return (tt1 == tt2);
         }
 
-        if (tt1.nInputs() != tt2.nInputs()) {
+        // both the truth table must have the same number of I/O combinations.
+        if (tt1.size() != tt2.size()) {
             return false;
         }
 
-        const auto nBits = tt1.nInputs();
-        assert(nBits < 65U);
-        const auto totalInputs = 1U << nBits;
+        // no. of primary outputs.
+        const auto noOfPrimaryOutputs = static_cast<int>(std::min(tt1.nOutputs(), tt2.nOutputs()));
```

Comment:
this does not seem like an appropriate check that makes sure the TTs have the same number of primary outputs.
Potentially, the TT class could use a member function `nPrimaryOutputs()` that counts the lines that are not garbage, as well as a `nPrimaryInputs()` function that counts the non-constant inputs.